cmd/history/history.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Extract git history from the main kata repos and generate two report files:
+#  report.txt - commits merged, sorted by repo section, and date.
+#  sorted_report.txt - all commits sorted by date
+#
+# This information can be useful when tracking down 'what changed when'.
+
+set -e
+
+REPORTFILE="report.txt"
+
+# Set a date to start from, or use the default of '1 week'
+#START="2019-03-08"
+START=$(date -I -d "last week")
+END=$(date -I)
+
+repo_base="github.com/kata-containers"
+repos="${repo_base}/runtime \
+	${repo_base}/proxy \
+	${repo_base}/shim \
+	${repo_base}/tests \
+	${repo_base}/agent \
+	${repo_base}/osbuilder \
+	${repo_base}/packaging"
+
+msg() {
+	local msg="$*"
+	echo "${msg}" | tee -a "${REPORTFILE}"
+}
+
+# Blank the file
+echo "" > "${REPORTFILE}"
+
+msg "===================================="
+msg "Commits from $START to $END"
+msg "===================================="
+for repo in $repos; do
+	repopath="${GOPATH}/src/$repo"
+	(go get -d -u ${repo} || true)
+	git -C "${repopath}" checkout master
+	git -C "${repopath}" pull >/dev/null
+	msg "---------- $repo ---------------"
+	TZ=UTC git -C "${repopath}" log --since "$START" --until "$END" --pretty="%h: %cd: ${repo##*/}: %s" --date=format:"%Y-%m-%dT%H:%M:%S" --no-merges | tee -a "${REPORTFILE}"
```

Comment:
You could avoid the `##*/` by building the github url inside the loop using an array. Something like:

```sh
repos=()
repos+=("runtime")
repos+=("shim")
  :

for repo in "${repos[@]}"
do
    github_repo="${repo_base}/${repo}"
    repopath="${GOPATH}/src/${github_repo}"
        :
  
done
```